src/lib/widget/BufFlatten.svelte

```diff
@@ -0,0 +1,176 @@
+<script>
+  // https://sashamaps.net/docs/resources/20-colors/
+  const colors = [[230, 25, 75], [60, 180, 75], [255, 225, 25], [0, 130, 200], [245, 130, 48], [70, 240, 240], [240, 50, 230], [250, 190, 212], [0, 128, 128], [220, 190, 255], [170, 110, 40], [255, 250, 200], [128, 0, 0], [170, 255, 195], [0, 0, 128], [128, 128, 128], [255, 120, 120], [0, 0, 0]]
+  const css_colors = colors.map(x => `rgb(${x[0]}, ${x[1]}, ${x[2]})`);
+  const shapes = [
+      '',
+      'polygon(0% 0%, 0% 0%, 100% 100%, 0% 100%)',
+      'polygon(50% 0%, 100% 50%, 50% 100%, 0% 50%)',
+      'circle(50%)',
+      'polygon(100% 100%, 0% 0%, 0% 100%, 100% 0%)',
+  ];
+  
+  let displayNumFrames = -1;
+  let displayNumChans = -1;
+
+  let srcFrames = 5;
+  let srcChans = 3;
+
+  let startFrame = 0;
+  let numFrames;
+
+  let startChan = 0;
+  let numChans;
+  
+  let axis = 1;
+  
+  let numFrames_error = '';
+  let numChans_error = ''; 
+    
+  $:{
+    numFrames_error = '';
+    numChans_error = '';
+    
+    if(displayNumFrames===0){
+      numFrames_error = 'numFrames cannot = 0';
+    } 
+    
+    if(displayNumChans===0){
+      numChans_error = 'numChans cannot = 0'
+    }
+    
+    numFrames = displayNumFrames===-1 ? srcFrames : displayNumFrames;
+    numChans = displayNumChans===-1 ? srcChans : displayNumChans;
+    
+    if(startFrame+numFrames > srcFrames){
+      startFrame_error = 'startFrame + numFrames cannot be > the number of frames in the soruce';
```

Comment:
This isn't defined outside the reactive statement so it will throw a breaking error to the console. It also seems that it's not actually being used in the markup either I would like to go through the code anyway and normalise some of the style so I'll add this in while I'm there

Reply:
oh yeah, i forgot about this guy. i ended up approaching this "warning" differently, but yes, putting it in would be good!